using-mobile-extensions/places-extension/places-monitoring-extension/using-your-own-monitor.md

```diff
@@ -1,4 +1,142 @@
 # Using your own monitor
 
-This information is  in progress.
+You can also use your own monitoring services and intergrate with Places using the Places Extension API's
 
+## Registering Geofences
+
+In the occasion of using your own monitoring services, register the geofences of the near by point of intrests around the current location by implementing the following steps.
+
+{% tabs %}
+
+{% tab title="iOS" %}
+
+1. The location updates obtained from the Core Location services of the iOS should be passed to the Places Extension. Use the `getNearbyPointsOfInterest` Places Extension API to get the array of n  `ACPPlacesPoi` objects around the current location.
+
+```objective-c
+- (void) locationManager: (CLLocationManager*) manager didUpdateLocations: (NSArray<CLLocation*>*) locations {
+
+        [ACPPlaces getNearbyPointsOfInterest:currentLocation limit:10 callback: ^ (NSArray<ACPPlacesPoi*>* _Nullable nearbyPoi) {
+            [self startMonitoringGeoFences:nearbyPoi];
+    }];
+
+}
+```
+
+2. Extract the information from the obtained `ACPPlacesPOI` objects and start monitoring those POI's
+
+```objective-c
+- (void) startMonitoringGeoFences: (NSArray*) newGeoFences {
+	// verify if the device supports monitoring geofences
+	// check for location permission
+
+    for (ACPPlacesPoi * currentRegion in newGeoFences) {
+        // make the circular region
+        CLLocationCoordinate2D center = CLLocationCoordinate2DMake(currentRegion.latitude, currentRegion.longitude);
+        CLCircularRegion* currentCLRegion = [[CLCircularRegion alloc] initWithCenter:center                                                                              												radius:currentRegion.radius                                                                     											   identifier:currentRegion.identifier];
+        currentCLRegion.notifyOnExit = YES;
+        currentCLRegion.notifyOnEntry = YES;
+
+        // start monitoring the new region
+        [_locationManager startMonitoringForRegion:currentCLRegion];
+
+    }
+}
+```
+
+
+
+{% endtab %}
+
+{% tab title="Android" %}
+
+-  The location updates obtained from the google play services or the android location services should be passed to the Places Extension. Use the `getNearbyPointsOfInterest` Places Extension API to get the list of n  `PlacesPoi` objects around the current location.
```

Comment:
- This text should not be in italics.

- Please change the sentences to read as follows:

"The location updates obtained from the Google Play services or the Android location services should be passed to the Places extension. To get the list of n  `PlacesPoi` objects around the current location, use `getNearbyPointsOfInterest`."